### PR TITLE
Task.51830

### PR DESCRIPTION
**Pull request by dangthibichngocb**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/51830)

## WHAT (optional)
- show list user in admin page

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2022-08-17 08-48-27

### CODE REVIEW

**Comment by dangthibichngocb** [comment]
ready

File: app/models/address.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class Address < ApplicationRecord
   belongs_to :user
   has_many :orders, dependent: :destroy
+
+  validates :user_id, presence: true
```
Comment (huongnt-2545) [inline app/models/address.rb]: bỏ dòng này đi em nhé, có association ở line 2 là cover đc case này rồi

**Comment by dangthibichngocb** [comment]
ready

File: app/models/user.rb [hunk 2]
```diff
@@ -7,6 +7,8 @@ class User < ApplicationRecord
     admin: Settings.const.users.role.admin
   }
 
+  enum status: {block: 2, active: 1}
```
Comment (ngocvt-0484) [inline app/models/user.rb]:
```suggestion
  enum status: {active: 1, block: 2}
```

File: app/views/admin/users/_user.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<% @users.each do |user| %>
```
Comment (ngocvt-0484) [inline app/views/admin/users/_user.html.erb]: bỏ dòng này
e dùng render collection là dc

File: app/views/admin/users/_user.html.erb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      <%= user.id %>
+    </td>
+    <td>
+      <%= user.name %>
+    </td>
+    <td>
+      <%= user.phone %>
+    </td>
+    <td>
+      <%= user.addresses.default_types.pluck(:name)[0] %>
```
Comment (ngocvt-0484) [inline app/views/admin/users/_user.html.erb]:
```suggestion
      <%= user.addresses.default_types.first.name %>
```

File: app/views/admin/users/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+              <th>
+                <%= t ".address" %>
+              </th>
+              <th>
+                <%= t ".status" %>
+              </th>
+            </tr>
+          </thead>
+          <tbody>
+            <%= render "user" %>
```
Comment (ngocvt-0484) [inline app/views/admin/users/index.html.erb]:
```suggestion
            <%= render @users %>
```

**Comment by dangthibichngocb** [comment]
ready

**Pull request merged**